crates/bitwarden/src/mobile/client_kdf.rs

```diff
@@ -0,0 +1,23 @@
+use crate::{error::Result, Client};
+
+use super::kdf::{get_user_password_hash, set_kdf_params, KdfParamRequest, PasswordHashRequest};
+
+pub struct ClientKdf<'a> {
+    pub(crate) client: &'a mut crate::Client,
+}
+
+impl<'a> ClientKdf<'a> {
+    pub async fn set_kdf_params(&mut self, req: KdfParamRequest) -> Result<()> {
+        set_kdf_params(self.client, req).await
+    }
+
+    pub async fn get_user_password_hash(&mut self, req: PasswordHashRequest) -> Result<String> {
+        get_user_password_hash(self.client, req).await
+    }
```

Comment:
Do we need to persist the kdf params or could they be an input to `get_user_password_hash`?

And should we rename it to `hash_password`.

Reply:
We use the KDF params both to create the master password hash for login, and to decrypt the symmetric encryption key in `InitUserCrypto`. 

The reason for the separation in two steps is for a vault unlock scenario, where you already have the vault and encryption keys locally. In those cases, you'd only need to call `set_kdf_params` and then initialize the crypto. But we could just require the KDF parameters as input for the crypto initialization as well.

Agreed on the rename.

Reply:
Hmm that's a valid point.

It does mean on say account switching they would need to call both `set_kdf_params` and `InitCrypto`, which might be a bit confusing.

Reply:
I've removed the set_kdf_params now, both options seem to have some pros and cons, but at least this way we reduce the API surface a bit